crates/p2p/src/client/peer_agnostic.rs

```diff
@@ -669,8 +672,8 @@ mod header_stream {
                     _ = tx.send(PeerData::new(peer, hdr)).await;
 
                     *start = match direction {
-                        Direction::Forward => *start + 1,
-                        Direction::Backward => *start - 1,
+                        Direction::Forward => start.saturating_add(1),
+                        Direction::Backward => start.saturating_sub(1),
```

Comment:
Using u64 & saturating sub here causes a loss in information and allows for receiving more headers than requested, which could be a dos vector. This is also why 2 tests fail currently, which check that we do not receive the next (not requested) header after the genesis one.

Reply:
Yeah, I've realized this too -- have reverted this part.